File: src/main/java/org/hypejet/hype/lang/PlayerSettingsListener.java
package org.hypejet.hype.lang;

import net.minestom.server.MinecraftServer;
import net.minestom.server.entity.Player;
import net.minestom.server.event.player.PlayerSettingsChangeEvent;

import java.util.HashMap;
import java.util.Locale;

final class PlayerSettingsListener {

    final HashMap<Player, Locale> locales = new HashMap<>();

    public void listener(PlayerSettingsChangeEvent event, LangApiImpl api) {
        if (!locales.containsKey(event.getPlayer())) {
            locales.put(event.getPlayer(), event.getPlayer().getLocale());
            MinecraftServer.getGlobalEventHandler().call(new PlayerLanguageChangeEvent(event.getPlayer(), event.getPlayer().getLocale(), true));
        }
        if(!locales.get(event.getPlayer()).equals(event.getPlayer().getLocale())) {
            locales.put(event.getPlayer(), event.getPlayer().getLocale());
            if (api.isClientLocale(event.getPlayer()))
                MinecraftServer.getGlobalEventHandler().call(new PlayerLanguageChangeEvent(event.getPlayer(), event.getPlayer().getLocale(), true));
        }
    }
}

File: src/main/kotlin/org/hypejet/hype/lang/PlayerSettingsListener.kt
package org.hypejet.hype.lang

import net.minestom.server.MinecraftServer
import net.minestom.server.entity.Player
import net.minestom.server.event.player.PlayerSettingsChangeEvent
import java.util.*

internal class PlayerSettingsListener {
    val locales = HashMap<Player, Locale?>()
    fun listener(event: PlayerSettingsChangeEvent, api: LangApiImpl) {
        if (!locales.containsKey(event.player)) {
            locales[event.player] = event.player.locale
            MinecraftServer.getGlobalEventHandler().call(
                PlayerLanguageChangeEvent(
                    event.player,
                    event.player.locale!!, true
                )
            )
        }
        if (locales[event.player] != event.player.locale) {
            locales[event.player] = event.player.locale
            if (api.isClientLocale(event.player)) MinecraftServer.getGlobalEventHandler().call(
                PlayerLanguageChangeEvent(
                    event.player,
                    event.player.locale!!, true
                )
            )
        }
    }
}